### ISSUE TITLE
nocatalog option

### ISSUE THREAD

**Issue by samuel4x4** [php-opencloud/openstack#237]
Hello,

Is there an option in the PHP SDK to get a token without the full catalog attached (like /auth/tokens?nocatalog)? I see Services uses $token->catalog, so an implementation should take this into consideration. 

If there's no 'nocatalog' option implemented, I would like to try implementing it. If you have any suggestions/remarks on this, please ley me know..

Thanks

**Comment by samuel4x4** [comment]
@haphan any idea? :)

**Comment by haphan** [comment]
@samuel4x4 What is the use-case of this feature request? Is it the overhead of sending extra data your concern?

**Comment by samuel4x4** [comment]
Yes, exactly that's the reason. Can we have have this implemented?

**Comment by haphan** [comment]
Again, can you give a sample use-case for querying auth token?

Implementation wise, if I am not wrong, OpenStack API server always responses with a catalog object. That means the overhead exists anyway.

**Comment by samuel4x4** [comment]
https://developer.openstack.org/api-ref/identity/v3/#password-authentication-with-unscoped-authorization

(Since v3.1) The authentication response excludes the service catalog. By default, the response includes the service catalog.

I see we should ask for the catalog only when we need it. It can stay included by default, as it is now, but to have the option to exclude it when you consider.

**Comment by haphan** [comment]
@samuel4x4 Since this option is now supported by Openstack API, I don't see any reason not to support it. 

If you already have implemented, I'd love to merge.

**Comment by samuel4x4** [comment]
@haphan I don't have yet a PR for this, I hope I'll find soon some time for this ;)
